File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/cdp/di/ecs/model/retailers/ECSRetailerList.java
/**
 * (C) Koninklijke Philips N.V., 2015.
 * All rights reserved.
 */
package com.philips.cdp.di.ecs.model.retailers;

import java.io.Serializable;
import java.util.List;

public class ECSRetailerList implements Serializable {

    private WrbresultsEntity wrbresults;

    public WrbresultsEntity getWrbresults() {
        return wrbresults;
    }

    List<ECSRetailer> getRetailers() {
        if (getWrbresults() != null && getWrbresults().getOnlineStoresForProduct() != null
                && getWrbresults().getOnlineStoresForProduct().getStores() != null && getWrbresults().getOnlineStoresForProduct().getStores().getRetailerList() != null) {

            return getWrbresults().getOnlineStoresForProduct().getStores().getRetailerList();
        }
        return null;
    }
}

File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/microService/model/retailer/ECSRetailerList.kt
/*
 *  Copyright (c) Koninklijke Philips N.V., 2020
 *
 *  * All rights are reserved. Reproduction or dissemination
 *
 *  * in whole or in part is prohibited without the prior written
 *
 *  * consent of the copyright holder.
 *
 *
 */

package com.philips.platform.ecs.microService.model.retailer

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ECSRetailerList(val wrbresults: Wrbresults? = null): Parcelable {

    fun getRetailers() : List<ECSRetailer>?{
      return wrbresults?.OnlineStoresForProduct?.Stores?.Store
    }

}
